File: prog14_.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

string longestCommonPrefix(vector<string>& strs)
{
    string prefix;
    if (strs.size() == 0)
        return prefix;

    bool match=true;
    for (int i=0;i<strs[0].size();i++)
    {
        char c=strs[0][i];
        for (int j=1;j<strs.size();j++)
        {
            if (c != strs[j][i])
                match=false;
        }
        if (match)
            prefix += c;
    }
    return prefix;
}

int main()
{
    vector<string> s={"abab","aba","abc"};
    cout << longestCommonPrefix(s) << endl;
    return 0;
}

File: prog14.py
def longestCommonPrefix(strs):
    prefix=""
    if len(strs)==0:
        return prefix

    lengths=[]
    for x in strs:
        lengths.append(len(x))
    n=min(lengths)
    match=True
    for i in range(n):
        c=strs[0][i]
        for j in range(len(strs)):
            if c != strs[j][i]:
                match=False

        if match:
            prefix += c
    return prefix

s=["abab","aba","abc"]
print (longestCommonPrefix(s))
